### ISSUE TITLE
Getting sentence level log probabilities / confidence scores

### ISSUE THREAD

**Issue by j-silv** [SYSTRAN/faster-whisper#1358]
Hello,

For my use case, I have audio files which are < 30 seconds with the following characteristics:

- they contain english speech with non-speech elements (car, dish washing, etc).
- multiple sentences with pauses in-between

I want to use the VAD filter so that I can get clean timestamps for when there is actual speech. I want to do an additional filtering step where if the sentence-level score (average log probability) is low, I want to discard that sentence from my training set.

Currently, faster-whisper only provides a segment level confidence score across the whole audio file:

https://github.com/SYSTRAN/faster-whisper/blob/c26d609974ef7c36715f23f0fbcdb3f9b5f8a663/faster_whisper/transcribe.py#L1449-L1452

[I know that if](https://github.com/SYSTRAN/faster-whisper/issues/232) `word_timestamps=True` we will get a `probability` value for each word, but is it appropriate to average up those probabilities across sentences and treat it as a "sentence level confidence score"? i.e. is it valid to do something like:

```
for segment in segments:
    logprobs = []
    print("[%.2fs -> %.2fs] %s (%.4f)" % (segment.start, segment.end, segment.text, segment.avg_logprob))
    for word in segment.words:
        print("[%.2fs -> %.2fs] %s (%.4f)" % (word.start, word.end, word.word, word.probability))
        logprobs.append(log(word.probability))
    avg_logprob = sum(logprobs) / len(logprobs)
    print(f"{avg_logprob=}")
```

With my audio file, this results in an output like:

```
... 
[6.31s -> 11.29s]  I don't like, I don't like how much sense that makes. (-0.4027)
[6.31s -> 6.31s]  I (0.9340)
[6.31s -> 7.37s]  don't (0.9953)
[7.37s -> 7.37s]  like, (0.9229)
[7.37s -> 8.75s]  I (0.6454)
[8.75s -> 9.03s]  don't (0.9970)
[9.03s -> 9.45s]  like (0.9955)
[9.45s -> 9.69s]  how (0.7275)
[9.69s -> 9.99s]  much (0.9949)
[9.99s -> 10.41s]  sense (0.9839)
[10.41s -> 10.69s]  that (0.9892)
[10.69s -> 11.29s]  makes. (0.9444)
avg_logprob=-0.0915

[14.79s -> 16.19s]  Okay, that's not a good question. (-0.4027)
[14.79s -> 15.05s]  Okay, (0.7655)
[15.37s -> 15.59s]  that's (0.9795)
[15.59s -> 15.75s]  not (0.2450)
[15.75s -> 15.97s]  a (0.2491)
[15.97s -> 16.01s]  good (0.9978)
[16.01s -> 16.19s]  question. (0.5408)
avg_logprob=-0.6168
```

If it's reasonable to use `probability` for each `Word` in this way, then perhaps my issue can be ignored. However, I feel like that value is too different from the token level logprob which would be obtained from the [generate function](https://opennmt.net/CTranslate2/python/ctranslate2.models.Whisper.html#ctranslate2.models.Whisper.generate). [CT2](https://github.com/OpenNMT/CTranslate2/pull/1489) can return this information.

I am happy to work on a PR such that the `generate` function returns the raw logits, and then we bundle those scores into a word-level prediction probability instead of just a probability for alignment. I'm just not sure whether or not that is adding additional benefit.
